task10_4/task10_4/Program.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+
+namespace task10_4
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Введите натуральное число, все цифры которого должны отличаться:");
+
+            int number;
+            if (!int.TryParse(Console.ReadLine(), out number) || number < 1)
+            {
+                Console.WriteLine("Число не натуральное");
+
+                return;
+            }
+
+            if (!int.TryParse(Console.ReadLine(), out number) || RepeatedDigits(number))
+            {
+                Console.WriteLine("Цифры в числе повторяются");
+
+                return;
+            }
```

Comment:
Программа не работает как надо. Вот листинг
```
Введите натуральное число, все цифры которого должны отличаться:
123456789

Цифры в числе повторяются
```
Ошибка в логическом выражении
```suggestion
            if (!int.TryParse(Console.ReadLine(), out number) && RepeatedDigits(number))
            {
                Console.WriteLine("Цифры в числе повторяются");

                return;
            }
```
Но даже после исправления этой ошибки ошибки получаем вот такое
````
Введите натуральное число, все цифры которого должны отличаться:
123456789

Позиция наибольшей цифры 0 в числе 0 равна 0
Позиция наименьшей цифры 0 в числе 0 равна 0
``